verification_token/models.py

```diff
@@ -19,14 +19,35 @@ class VerificationTokenManager(models.Manager):
     def deactivate(self, obj, slug=None, key=None):
         self.filter_active_tokens(obj, slug, key).update(is_active=False)
 
-    def deactivate_and_create(self, obj, slug=None, extra_data=None, deactivate_old_tokens=True, expiration_in_minutes=None,
-                              key_generator_kwargs=None):
-        expiration_in_minutes = settings.DEFAULT_EXPIRATION if expiration_in_minutes is None else expiration_in_minutes
+    def deactivate_and_create(self, obj, slug=None, extra_data=None, deactivate_old_tokens=True,
+                              key_generator_kwargs=None, **kwargs):
```

Comment:
Trochu brutus s tšmi argumenty a kwargy, ale jediný způsob, který mě napadl, jak zachovat default hodnotu a zpětnou kompatibilitu, pokus není hodnota expiration_in_minutes výslovně uvedena a současně umožnit použití expiration_in_minutes=None jako validní samostatné hodnoty.

Samozřejmě, pokud někdo výslovně uváděl hodnotu None a čekal default, pak má smůlu, ale tomu se zřejmě nelze vyhnout - je to buď a nebo.

Reply:
Asi bych udělal speciální konstantu INFINITY, kterou bys mohl použít. Může mít hodnotu třeba -1. Případně proč nedáš jako default tu hodnotu ze settings, místo None?

Reply:
A pak bude (s hodnotou -1) token expirovat dřív, než začal platit, nebo se bude předělávat logika validity a platnosti?

Reply:
Ad 'Případně proč nedáš jako default tu hodnotu ze settings, místo None?' - protože pak selže test - že lze hodnotu nastavit pomocí settings.

Reply:
Tu hodnotu nebudeš nastavovat do toho modelu, je to jen rozlišovák jestli nastavit default nebo None.

Reply:
Té druhé poznámce nerozumím, na čem to fláklo?

Reply:
souhlasím s Ondrou lepší než none bude nějaká konstanta

Reply:
doporučuji podobně jako to dělá django

Reply:
class Empty:
    pass

EMPTY = Empty()

Reply:
samozřejmě lépe pojmenované a do modelu ukládat None v tomto případě. Defaultní hodnotu na vstupu použít ze settings pokud nebude nastavena, ale dělej to přes ten pop jinak to nebude dynamické a nepůjde to testovat

Reply:
po dohodě s @matllubos - konstanta je little improvement a dá se udělat i později - takže teď ne.